src/EditorFeatures/CSharpTest/NavigateTo/NavigateToSearcherTests.cs

```diff
@@ -23,35 +23,71 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.NavigateTo
     [Trait(Traits.Feature, Traits.Features.NavigateTo)]
     public class NavigateToSearcherTests
     {
-        private static IReturnsResult<INavigateToSearchService> SetupSearchProject(
+        private static void SetupSearchProject(
             Mock<INavigateToSearchService> searchService,
             string pattern,
             bool isFullyLoaded,
             INavigateToSearchResult? result)
         {
-            var searchServiceSetup = searchService.Setup(ss => ss.SearchProjectAsync(
-                It.IsAny<Project>(),
-                It.IsAny<ImmutableArray<Document>>(),
-                pattern,
-                ImmutableHashSet<string>.Empty,
-                isFullyLoaded,
-                It.IsAny<Func<INavigateToSearchResult, Task>>(),
-                It.IsAny<CancellationToken>()));
-
-            var searchServiceCallback = searchServiceSetup.Callback(
-                (Project project,
-                 ImmutableArray<Document> priorityDocuments,
-                 string pattern2,
-                 IImmutableSet<string> kinds,
-                 bool isFullyLoaded2,
-                 Func<INavigateToSearchResult, Task> onResultFound2,
-                 CancellationToken cancellationToken) =>
-                {
-                    if (result != null)
-                        onResultFound2(result);
-                });
-
-            return searchServiceCallback.Returns(Task.CompletedTask);
+            if (isFullyLoaded)
+            {
+                // First do a full search
+                searchService.Setup(ss => ss.SearchProjectAsync(
```

Comment:
it looks like we're replicating the ordering logic in the test - would it be possible to instead call `SearchAsync` and verify that the projects / documents are all searched in the correct order?  If not for all tests, maybe just one new one?

It is kind of an implementation detail, but one that is visible to the user (slower results)